scheduler/rank.go

```diff
@@ -285,8 +285,36 @@ OUTER:
 			for _, req := range task.Resources.Devices {
 				offer, sumAffinities, err := devAllocator.AssignDevice(req)
 				if offer == nil {
-					iter.ctx.Metrics().ExhaustedNode(option.Node, fmt.Sprintf("devices: %s", err))
-					continue OUTER
+					// If eviction is not enabled, mark this node as exhausted and continue
```

Comment:
Known issue - binpack iterator code flow is too complex to follow. I played with some ideas for refactoring as part of this PR but didn't land on anything reasonable. I'll make a separate PR that simplifies` BinPackIterator`